src/shape.rs

```diff
@@ -660,237 +677,214 @@ impl ShapeLine {
+                    );
                 } else {
                     // congruent direction
                     let mut fitting_start = (0, 0);
                     for (i, word) in span.words.iter().enumerate() {
-                        let word_size = font_size * word.x_advance;
-                        if fit_x - word_size >= 0. {
+                        let word_width = font_size * word.x_advance;
+                        if fit_x - word_width >= 0. {
                             // fits
-                            fit_x -= word_size;
-                            word_range_width += word_size;
+                            fit_x -= word_width;
+                            word_range_width += word_width;
                             if word.blank {
                                 number_of_blanks += 1;
                             }
                             continue;
                         } else if wrap == Wrap::Glyph {
                             for (glyph_i, glyph) in word.glyphs.iter().enumerate() {
-                                let glyph_size = font_size * glyph.x_advance;
-                                if fit_x - glyph_size >= 0. {
-                                    fit_x -= glyph_size;
-                                    word_range_width += glyph_size;
+                                let glyph_width = font_size * glyph.x_advance;
+                                if fit_x - glyph_width >= 0. {
+                                    fit_x -= glyph_width;
+                                    word_range_width += glyph_width;
                                     continue;
                                 } else {
-                                    word_ranges.push((
+                                    self.add_to_visual_line(
+                                        &mut current_visual_line,
+                                        span_index,
                                         fitting_start,
                                         (i, glyph_i),
                                         word_range_width,
                                         number_of_blanks,
-                                    ));
+                                    );
+                                    visual_lines.push(current_visual_line);
+                                    current_visual_line = VisualLine::default();
+
                                     number_of_blanks = 0;
-                                    fit_x = line_width - glyph_size;
-                                    word_range_width = glyph_size;
+                                    fit_x = line_width - glyph_width;
+                                    word_range_width = glyph_width;
                                     fitting_start = (i, glyph_i);
                                 }
                             }
                         } else {
                             // Wrap::Word
                             let mut prev_word_width = None;
```

Comment:
I think for better consistency, we should rename this to `trailing_space_width` too.

Reply:
Fixed in [7fa51c6](https://github.com/pop-os/cosmic-text/pull/108/commits/7fa51c6404f7f286a4c23abc79640bc0e63a5138)